include/exadg/fluid_structure_interaction_precice/driver_solid.h

```diff
@@ -0,0 +1,333 @@
+                                     *this->application->get_grid_structure()->mapping);
+
+      this->timer_tree.insert({"FSI", "Setup", "Structure"}, timer_local.wall_time());
+    }
+    /**************************************** STRUCTURE *****************************************/
+
+    /**************************************** STRUCTURE *****************************************/
+
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+    this->precice =
+      std::make_shared<Adapter::Adapter<dim, dim, VectorType>>(this->precice_parameters);
+
+    write_mesh_name = this->precice_parameters.write_mesh_name;
+    this->precice->add_write_interface(
+      this->application->get_boundary_descriptor_structure()->neumann_mortar_bc.begin()->first,
+      this->precice_parameters.write_mesh_name,
+      {this->precice_parameters.write_data_name, "Velocity"},
+      "values_on_dofs",
+      structure_matrix_free,
+      structure_operator->get_dof_index(),
+      numbers::invalid_unsigned_int);
+
+    // structure to ALE
+    // structure to fluid
+
+    // fluid to structure
+    {
+      std::vector<unsigned int> quad_indices;
+      quad_indices.emplace_back(structure_operator->get_quad_index());
+
+      // VectorType stress_fluid;
+      communicator = std::make_shared<InterfaceCoupling<dim, dim, Number>>(this->precice);
+      VectorType displacement_structure;
+      structure_operator->initialize_dof_vector(displacement_structure);
+      communicator->setup(structure_matrix_free,
+                          structure_operator->get_dof_index(),
+                          quad_indices,
+                          this->application->get_boundary_descriptor_structure()->neumann_mortar_bc,
+                          this->precice_parameters.read_mesh_name,
+                          "Stress");
+
+      displacement_structure = 0;
+      this->precice->initialize_precice(displacement_structure);
+    }
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+
+    /**************************** SETUP TIME INTEGRATORS AND SOLVERS ****************************/
+
+    // Structure
+    {
+      // initialize time integrator
+      structure_time_integrator = std::make_shared<Structure::TimeIntGenAlpha<dim, Number>>(
+        structure_operator,
+        structure_postprocessor,
+        this->application->get_parameters_structure(),
+        this->mpi_comm,
+        this->is_test);
+
+      structure_time_integrator->setup(
+        this->application->get_parameters_structure().restarted_simulation);
+
+      structure_operator->setup_solver();
+    }
+
+    /**************************** SETUP TIME INTEGRATORS AND SOLVERS ****************************/
+  }
+
+  void
+  solve() const final
+  {
+    Assert(this->application->get_parameters_fluid().adaptive_time_stepping == false,
```

Comment:
should probably be `get_parameters_solid()` since this is the solid driver?

I assume it works currently because you call `application->setup()` above, which sets up everything. However, in the future, one should introduce function `setup_structure()` and `setup_fluid_and_ale()` in `ApplicationBase`. In that case, the parameter accessed above would be invalid. For now, we should make sure that this bug does not occur later.